dll/win32/browseui/globalfoldersettings.cpp

```diff
@@ -20,6 +20,35 @@
 
 #include "precomp.h"
 
+#define ROSSIG ( ('R' << 0) | ('O' << 8) | ('S' << 16) | (('g' ^ 'f' ^ 's') << 24) )
+#define DEFBROWSERSTREAM L"Settings"
+
+template<class S, class D> static void CopyTo(const S &Src, D &Dst)
+{
+    Dst.FolderSettings = Src.FolderSettings;
+}
+
+static void EnsureValid(FOLDERSETTINGS &fs)
+{
+    if ((int)fs.ViewMode < FVM_AUTO || (int)fs.ViewMode > FVM_LAST)
+        fs.ViewMode = SBFOLDERSETTINGS::DEF_FVM;
+
+    fs.fFlags &= ~(FWF_OWNERDATA | FWF_DESKTOP | FWF_TRANSPARENT | FWF_NOCLIENTEDGE |
+                   FWF_NOSUBFOLDERS | FWF_NOSCROLL | FWF_NOENUMREFRESH | FWF_NOVISIBLE);
+}
+
+static void EnsureValid(DEFFOLDERSETTINGS &dfs)
```

Comment:
Is this EnsureValid helper meant to be used in more than one place in the future? (So far it's only used in CGlobalFolderSettings::Load )

Reply:
No sure yet. Not in this PR anyway.